### ISSUE TITLE
Operators without xuid?

### ISSUE THREAD

**Issue by RealTadango** [itzg/docker-minecraft-bedrock-server#378]
I want to setup a server for my kids but when they login there is no xuid in the log. They don't have mojang accounts, just a local profile in the app. I want them to be operator in the server when they login. What can i do?

**Comment by itzg** [comment]
I thought Microsoft/Mojang/Xbox account was required to use any of the Minecraft mobile/console apps. In which case, I am not familiar with the "offline" access you're attempting and recommend you search more broadly for answers to that scenario.

Out of curiosity, can you post the server logs during one of the logins?

**Comment by RealTadango** [comment]
No, a mojang account is not needed :) Not on Android and the Playstation 4 where we use them. We can play together on the local network without an account. I am adding a docker server so they have a common server and don't rely on each others machine to start.

I found that the default user level set to operator works, but only when they have not joined before. the OP command is not "sticky" on the server. Sadly now all are there friends will also be operators when they come over and play... but at least they can have fun on the server.

**Comment by itzg** [comment]
Can you provide logs?

**Comment by itzg** [comment]
In any case it sounds like you have found the solution. There's nothing special the container setup can do beyond what you have done so far.

**Comment by RealTadango** [comment]
2023-12-05 19:07:50:757 INFO] Player connected: Zonnig, xuid: 
[2023-12-05 19:07:54:474 INFO] Player Spawned: Zonnig xuid: , pfid: 99d7a8123147739a
[2023-12-05 19:08:56:676 INFO] Player disconnected: Zonnig, xuid: , pfid: 99d7a8123147739a
[2023-12-05 19:10:17:664 INFO] Player connected: Zonnig, xuid: 
[2023-12-05 19:10:21:195 INFO] Player Spawned: Zonnig xuid: , pfid: 99d7a8123147739a
[2023-12-05 19:10:32:470 INFO] Player disconnected: Zonnig, xuid: , pfid: 99d7a8123147739a
[2023-12-05 19:11:54:215 INFO] Player connected: Zonnig, xuid: 
[2023-12-05 19:11:57:869 INFO] Player Spawned: Zonnig xuid: , pfid: 99d7a8123147739a
[2023-12-05 19:12:34:578 INFO] Player disconnected: Zonnig, xuid: , pfid: 99d7a8123147739a

This is what i see when i connect. No xuid but pfid is the same. It matches what i see in the app

**Comment by mattholy** [comment]
`xuid` is so called Xbox user id, if you didn't login so it will be `null`.

see https://www.roxservers.com/knowledgebase.php?action=displayarticle&id=835&language=english

**Comment by aerolincoln** [comment]
@RealTadango how did you end up solving this for your kids with the default permission? I am in the same situation you are and changed the DEFAULT_PLAYER_PERMISSION_LEVEL to "operator" but I am unable to use the teleport command in-game. My kid's character is off somewhere and I am trying to get us back together.

**Comment by RealTadango** [comment]
@aerolincoln Check if you see default-player-permission-level=operator in the server.properties. Also check you level after connecting.

**Comment by aerolincoln** [comment]
![image](https://github.com/itzg/docker-minecraft-bedrock-server/assets/88894291/b5be7d6e-4e4e-4264-984d-c97f47d12053)

I can't figure out how to check my own level in the game. The command /permission doesn't work perhaps because I don't have enough permission to use it.

**Comment by aerolincoln** [comment]
I'm guessing because I had already joined the server that the "default for new players" value didn't apply to me and I have to somehow flush that record for this to take effect.

**Comment by RealTadango** [comment]
> I'm guessing because I had already joined the server that the "default for new players" value didn't apply to me and I have to somehow flush that record for this to take effect.

I don't think so? Since there is no xuid it worked fine for me but if you have a xuid it might be different.

**Comment by aerolincoln** [comment]
No XUID for any user on the server.

**Comment by aerolincoln** [comment]
Resetting or restarting the server didn't work but deleting it and recreating it did. I saw some comment on another issue that suggests the permissions are not being properly set after the first time a player joins. I have access to all the operator commands now. Fortunately we hadn't been playing very long and didn't lose much.
